### DISCUSSION TITLE
new to github

### DISCUSSION THREAD

**Discussion by c2995509** [community/community#73534] [New to GitHub]
I'm wondering if it is possible to commit a folder instead of a file, As I have two files and may expand, Therefore I was wondering if it is possible to load files by folder instead of doing individual files?
i.e Folder a contains py files, text file and will contain a readme file

**Comment by jzelAdmin2006** [comment]
Hi @c2995509,

Absolutely, you can commit a folder to your GitHub repository, but keep in mind that Git, the version control system that underpins GitHub, tracks the contents of files rather than the folders themselves. This means that if you want to commit a folder, it must have at least one file in it.

Here's a step-by-step guide to help you get your folder with its files onto GitHub:

1. **Create your folder locally** and place your `.py` files, `.txt` file, and any other files you want to include in this folder.

2. **Open your terminal** (or Git Bash if you're on Windows) and navigate (`cd`) to your project directory.

3. **Initialize a new Git repository** with `git init`, if you haven't already done so.

4. **Add your folder to the staging area** with the command:
   ```sh
   git add FolderName/
   ```
   Make sure to replace `FolderName` with the name of your actual folder.

5. **Commit your changes** using:
   ```sh
   git commit -m "Added FolderName with necessary files"
   ```

6. **Create a new repository on GitHub.** After logging into GitHub, go to the Repositories tab and click the 'New' button.

7. **Link your local repository with the one on GitHub** using:
   ```sh
   git remote add origin https://github.com/YourUsername/YourRepositoryName.git
   ```
   Replace `YourUsername` and `YourRepositoryName` with your actual GitHub username and the new repository name, respectively.

8. **Push your commits to GitHub** using:
   ```sh
   git push -u origin main
   ```
   (Note: The default branch name on GitHub used to be `master`, but it has been changed to `main` on new repositories. Use the branch name that your repository has.)

In case your folder is empty or you want to commit an empty folder, you might want to add a `.gitkeep` file inside it since Git does not track empty folders. Just create an empty file named `.gitkeep` inside your folder and repeat the steps above.

I hope this clarifies the process for you. If you need any more help, feel free to reach out! Please mark this as an answer if you found it helpful.

Best regards
@jzelAdmin2006

**Comment by EliasAfara** [comment]
You definitely can.

**Comment by luzlloveras** [comment]
You also can go to Add Files and then Upload Files (folders or just documents separated).
